### PR TITLE
Add enterprise support for auth

### PR DESCRIPTION
**Pull request by djbarnwal**
Add Rill enterprise related login to the auth template.
cc @himadrisingh

### CODE REVIEW

**Comment by ericpgreen2** [comment]
We should get the design on this right before merging. Between Rill Cloud and existing Rill Enterprise, very quickly a lot of people will see this. I've nudged Janet to prioritize the Figma representation.

File: web-auth/README.md
Comment (ericpgreen2) [inline web-auth/README.md:1]: It'd be helpful if you said where a team member can find the `.env` file

File: web-auth/src/components/Auth.svelte [hunk 1]
```diff
@@ -11,18 +11,31 @@
...
 
+  $: loginOptions.forEach((option) => {
+    if (option.name === "Okta") {
+      option.connection = oktaName;
+    }
+    if (option.name === "Pingfed") {
+      option.connection = pingFedName;
+    }
+  });
+
```
Comment (ericpgreen2) [inline web-auth/src/components/Auth.svelte:38]: Can we just import the environment variables into `config.js` directly? (Btw, it'd be nice if that file were Typescript)

File: web-auth/src/routes/+page.svelte [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+    .VITE_DISABLE_FORGOT_PASS_DOMAINS;
 </script>
 
-<Auth {cloudClientIDs} configParams="@@config@@" />
+<Auth
+  {cloudClientIDs}
+  {oktaName}
+  {pingFedName}
+  {disableForgotPassDomains}
+  configParams="@@config@@"
```
Comment (ericpgreen2) [inline web-auth/src/routes/+page.svelte:16]: Why is `configParams` passed in as a prop when it's a constant?

Comment (ericpgreen2) [reply web-auth/src/routes/+page.svelte:16]: `configParams` is also a pretty vague name. I'm having a hard time following what this is.

File: web-auth/src/components/Auth.svelte [hunk 3]
```diff
@@ -11,18 +11,31 @@
...
   function initConfig() {
     const config = JSON.parse(
```
Comment (ericpgreen2) [inline web-auth/src/components/Auth.svelte:40]: I'm having a hard time following what "config" is

**Comment by ericpgreen2** [comment]
I tried to test this by removing the `if` statement around:
```typescript
loginOptions = loginOptions.filter(
  (option) => !["Okta", "Pingfed"].includes(option.name)
);
```

I'm still seeing all the login options. Am I doing something wrong in my dev/testing environment? Or is this a bug?

File: web-auth/README.md
Comment (djbarnwal) [reply web-auth/README.md:1]: Added in README

**Comment by ericpgreen2** [comment]
Thanks, now this README is so much more helpful! A couple things:
- I think it's better to put the `.env` file in 1Password, not in Notion. We use 1Password for Rill Cloud's `.env`. Can you do that & update the README?
- In the last section, you say "deploy the latest build to staging so the reviewer can test it out". Can you please add instructions for deploying an unmerged PR to staging?

**Comment by ericpgreen2** [comment]
This is looking good to me!

One thing for this PR:
- I see the term "connection" as in `connectionMap`, `connectionName`, `getConnectionFromEmail`, etc. The term is pretty vague. Is this Auth0 vernacular? Is a correct and more descriptive term "identity provider" / IDP? Or something else?

For the future, two improvements we've discussed:
- The long `state` parameter in the URL is tough to look at
- Two separate routes: one for `/login` and one for `/signup`

I don't 100% know the requirements for this PR, so @himadrisingh would you be able to do a final QA to make sure this PR adds the needed functionality?

**Comment by himadrisingh** [comment]
The login page works as expected.

**Comment by ericpgreen2** [comment]
> Yes `connection` is part of Auth0 vernacular used to represent enterprise login realms.

Ok, sounds good. Approving the PR 👍

**Review by ericpgreen2 — APPROVED** [review]

**Pull request merged**